File: app/src/test/java/com/github/goldy1992/mp3player/service/library/content/request/ContentRequestParserTest.java
package com.github.goldy1992.mp3player.service.library.content.request;


import com.github.goldy1992.mp3player.commons.MediaItemType;
import com.github.goldy1992.mp3player.service.library.MediaItemTypeIds;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static com.github.goldy1992.mp3player.commons.Constants.ID_SEPARATOR;
import static org.junit.jupiter.api.Assertions.assertEquals;

class ContentRequestParserTest {

    private ContentRequestParser contentRequestParser;


    private MediaItemTypeIds mediaItemTypeIds;


    @BeforeEach
    public void setup() {
        this.mediaItemTypeIds = new MediaItemTypeIds();
        this.contentRequestParser = new ContentRequestParser(mediaItemTypeIds);
    }

    @Test
    void testRootItem() {
        final String foldersId = mediaItemTypeIds.getId(MediaItemType.FOLDERS);
        final String folderId = mediaItemTypeIds.getId(MediaItemType.FOLDER);
        ContentRequest contentRequest = contentRequestParser.parse(foldersId);
        assertEquals(foldersId, contentRequest.getContentRetrieverKey());
        assertEquals(folderId, contentRequest.getMediaIdPrefix()); // each child of FOLDERS is of type FOLDER
        assertEquals(foldersId, contentRequest.getQueryString());
    }

    @Test
    void testGetFolderSongs() {
        final String folderId = mediaItemTypeIds.getId(MediaItemType.FOLDER);
        final String path = "/a/b/mediaPath";
        String id = folderId + ID_SEPARATOR + path;
        ContentRequest contentRequest = contentRequestParser.parse(id);
        assertEquals(folderId, contentRequest.getContentRetrieverKey());
        assertEquals(id, contentRequest.getMediaIdPrefix());
        assertEquals(path, contentRequest.getQueryString());
    }
}

File: service/src/test/java/com/github/goldy1992/mp3player/service/library/content/request/ContentRequestParserTest.kt
package com.github.goldy1992.mp3player.service.library.content.request

import com.github.goldy1992.mp3player.commons.Constants.ID_SEPARATOR
import com.github.goldy1992.mp3player.commons.MediaItemType
import com.github.goldy1992.mp3player.service.library.MediaItemTypeIds
import org.junit.Assert.assertEquals
import org.junit.Before
import org.junit.Test



class ContentRequestParserTest {
    private var contentRequestParser: ContentRequestParser? = null
    private var mediaItemTypeIds: MediaItemTypeIds? = null
    @Before
    fun setup() {
        mediaItemTypeIds = MediaItemTypeIds()
        contentRequestParser = ContentRequestParser(mediaItemTypeIds!!)
    }

    @Test
    fun testRootItem() {
        val foldersId = mediaItemTypeIds!!.getId(MediaItemType.FOLDERS)
        val folderId = mediaItemTypeIds!!.getId(MediaItemType.FOLDER)
        val contentRequest = contentRequestParser!!.parse(foldersId!!)
        assertEquals(foldersId, contentRequest!!.contentRetrieverKey)
        assertEquals(folderId, contentRequest.mediaIdPrefix) // each child of FOLDERS is of type FOLDER
        assertEquals(foldersId, contentRequest.queryString)
    }

    @Test
    fun testGetFolderSongs() {
        val folderId = mediaItemTypeIds!!.getId(MediaItemType.FOLDER)
        val path = "/a/b/mediaPath"
        val id = folderId + ID_SEPARATOR + path
        val contentRequest = contentRequestParser!!.parse(id)
        assertEquals(folderId, contentRequest!!.contentRetrieverKey)
        assertEquals(id, contentRequest.mediaIdPrefix)
        assertEquals(path, contentRequest.queryString)
    }
}